config/config_test.go

```diff
@@ -5,171 +5,211 @@ import (
+
+func (s *baseTestSuite) TestBaseNewFTWRegexpText() {
+	ftwRegex, err := NewFTWRegexp("test")
+	s.NoError(err)
+	s.NotNil(ftwRegex)
+	s.True(ftwRegex.MatchString("This is a test"), "looks like we could not match string")
+}
 
-	overrides := cfg.TestOverride.Overrides
-	assert.NotNil(t, overrides.DestAddr, "Looks like we are not overriding destination address")
-	assert.Equal(t, "httpbingo.org", *overrides.DestAddr, "Looks like we are not overriding destination address")
+func (s *baseTestSuite) TestNewCloudConfig() {
+	cfg := NewCloudConfig()
+	s.Equal(CloudRunMode, cfg.RunMode)
+	s.Equal("", cfg.LogFile)
 }
 
-func TestNewConfigBadConfig(t *testing.T) {
-	filename, _ := utils.CreateTempFileWithContent(yamlBadConfig, "test-*.yaml")
+func (s *baseTestSuite) TestNewDefaultConfig() {
+	cfg := NewDefaultConfig()
+	s.Equal(DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName)
+	s.Equal(DefaultRunMode, cfg.RunMode)
+	s.Equal("", cfg.LogFile)
+}
+
+func (s *fileTestSuite) TestNewConfigBadFileConfig() {
+	filename, _ := utils.CreateTempFileWithContent(testData["jsonConfig"], "test-*.yaml")
 	defer os.Remove(filename)
 	cfg, err := NewConfigFromFile(filename)
+	s.NoError(err)
+	s.NotNil(cfg)
+}
+
+func (s *fileTestSuite) TestNewConfigFromFile() {
+	s.NotEmpty(s.cfg.TestOverride.Overrides, "Ignore list must not be empty")
+
+	for id, text := range s.cfg.TestOverride.Ignore {
+		s.Contains((*regexp.Regexp)(id).String(), "920400-1$", "Looks like we could not find item to ignore")
+		s.Equal("This test must be ignored", text, "Text doesn't match")
+	}
+
+	overrides := s.cfg.TestOverride.Overrides
+	s.NotNil(overrides.DestAddr, "Looks like we are not overriding destination address")
+	s.Equal("httpbingo.org", *overrides.DestAddr, "Looks like we are not overriding destination address")
+}
 
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
+func (s *fileTestSuite) TestNewConfigBadConfig() {
+	// contents come from yamlBadConfig
+	s.NotNil(s.cfg)
 }
 
-func TestNewConfigDefaultConfig(t *testing.T) {
+func (s *fileTestSuite) TestNewConfigDefaultConfig() {
 	// For this test we need a local .ftw.yaml file
-	fileName := ".ftw.yaml"
-	_ = os.WriteFile(fileName, []byte(yamlConfig), 0644)
-	t.Cleanup(func() {
-		os.Remove(fileName)
-	})
+	s.filename = ".ftw.yaml"
+	_ = os.WriteFile(s.filename, []byte(testData["ok"]), 0644)
 
 	cfg, err := NewConfigFromFile("")
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
+	s.NoError(err)
+	s.NotNil(cfg)
 }
 
-func TestNewConfigFromString(t *testing.T) {
-	cfg, err := NewConfigFromString(yamlConfig)
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
+func (s *fileTestSuite) TestNewConfigFromString() {
+	cfg, err := NewConfigFromString(testData["ok"])
+	s.NoError(err)
+	s.NotNil(cfg)
 }
 
-func TestNewEnvConfigFromString(t *testing.T) {
-	cfg, err := NewConfigFromString(yamlConfig)
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
+func (s *fileTestSuite) TestNewEnvConfigFromString() {
+	cfg, err := NewConfigFromString(testData["ok"])
```

Comment:
Should test env, not string. Or should this test be deleted?

Reply:
Removed.